source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -1186,24 +1215,48 @@ inline void ComponentInterface<NodeT>::publish_transforms(
   }
 }
 
+
 template<class NodeT>
-inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_transform(
+inline geometry_msgs::msg::TransformStamped ComponentInterface<NodeT>::lookup_ros_transform(
     const std::string& frame, const std::string& reference_frame, const tf2::TimePoint& time_point,
     const tf2::Duration& duration
 ) {
   if (this->tf_buffer_ == nullptr || this->tf_listener_ == nullptr) {
     throw exceptions::LookupTransformException("Failed to lookup transform: To TF buffer / listener configured.");
   }
   try {
-    state_representation::CartesianPose result(frame, reference_frame);
-    auto transform = this->tf_buffer_->lookupTransform(reference_frame, frame, time_point, duration);
-    modulo_core::translators::read_message(result, transform);
-    return result;
+    return this->tf_buffer_->lookupTransform(reference_frame, frame, time_point, duration);;
   } catch (const tf2::TransformException& ex) {
     throw exceptions::LookupTransformException(std::string("Failed to lookup transform: ").append(ex.what()));
   }
 }
 
+
+template<class NodeT>
+inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_transform(
+    const std::string& frame, const std::string& reference_frame, const tf2::TimePoint& time_point,
+    const tf2::Duration& duration
+) {
+  auto transform = this->lookup_ros_transform(frame, reference_frame, time_point, duration);
+  state_representation::CartesianPose result(frame, reference_frame);
+  modulo_core::translators::read_message(result, transform);
+  return result;
+}
+
+template<class NodeT>
+inline state_representation::CartesianPose ComponentInterface<NodeT>::lookup_transform(
+    const std::string& frame, const std::string& reference_frame, double validity_period, const tf2::Duration& duration
+) {
+  auto transform =
+      this->lookup_ros_transform(frame, reference_frame, tf2::TimePoint(std::chrono::microseconds(0)), duration);
+  if (validity_period > 0.0 && (this->get_clock()->now() - transform.header.stamp).seconds() > validity_period) {
```

Comment:
See my comment above, this would actually avoid the conversion of the `validity_period` was a proper duration.